src/modern/Resolver.js

```diff
@@ -81,35 +83,44 @@ export default class Resolver {
     });
   }
 
-  getOperations(queries, routeVariables) {
+  updateQuerySubscriptions(queries, routeVariables, cacheConfigs) {
     const { createOperationSelector, getOperation } =
       this.environment.unstable_internal;
 
-    return queries.map((query, i) => {
+    const querySubscriptions = queries.map((query, i) => {
       if (!query) {
-        // TODO: Warn if variables are specified?
         return null;
       }
 
-      return createOperationSelector(getOperation(query), routeVariables[i]);
+      const variables = routeVariables[i];
+
+      if (
+        this.lastQueries[i] === query &&
+        areEqual(this.lastRouteVariables[i], variables)
```

Comment:
am I better off with `_.isEqual`? relay upstream uses `fbjs/lib/areEqual` though, so this would guarantee an exact match, but calling into fbjs isn't ideal